### PR TITLE
Added unit tests

### PR DESCRIPTION
**Pull request by jujugrrr**

### CODE REVIEW

**Comment by martinb3** [comment]
It looks like you've dropped the validation against cloud servers. We did that to mirror managed cloud automation, which also makes this check, because dedicated servers are typically patched through other means. Should we document this so that users don't get a surprise on dedicated gear?

File: Rakefile [hunk 1]
```diff
@@ -12,8 +12,8 @@ namespace :style do
   desc 'Run Chef style checks'
   FoodCritic::Rake::LintTask.new(:chef) do |t|
     t.options = { search_gems: true,
-                  fail_tags: %w(correctness rackspace),
-                  chef_version: '11.6.0'
+                  tags: %w(~rackspace-support),
+                  fail_tags: %w(correctness,rackspace)
```
Comment (martinb3) [inline Rakefile]: This creates an array with one element: `['correctness,rackspace']`. I think that's probably not right.

e.g.

```
irb(main):001:0> %w(a,b)
=> ["a,b"]
irb(main):002:0> %w(a b)
=> ["a", "b"]
```

Where did this Rakefile content come from that's broken?

**Comment by martinb3** [comment]
I feel like this is using a different Rakefile than many of the other projects. Did someone make some updates to the Rakefile and not update it everywhere? I feel like the most current one was on [platformstack](https://github.com/rackspace-cookbooks/platformstack/blob/master/Rakefile), but that doesn't match this one.

IIRC we discovered there was no point doing a `kitchen destroy` at the end vs. a `kitchen test`, since the separate destroy wasn't cleaning up any more failed servers than a `kitchen test`. We ended up adding the cleanup job to get those; they were mostly failing at the create setup where kitchen couldn't destroy them later anyway because it didn't know about them.

**Comment by martinb3** [comment]
Commit message should probably be updated to describe the changes, since it appears that this PR does more than `Added unit tests`. It appears to drop poise and also remove some cloud and OS validation checks.

File: Rakefile (hunk 1 above)
Comment (jujugrrr) [inline Rakefile]: I've just copied paste it. Which project do you want me to copy it form ?

**Comment by jujugrrr** [comment]
> It looks like you've dropped the validation against cloud servers. We did that to mirror managed cloud automation, which also makes this check, because dedicated servers are typically patched through other means. Should we document this so that users don't get a surprise on dedicated gear?

Hi,
Indeed, I wanted to talk about this. This libraries seems/(is supposed) to be generic, I don't see why we should not patch if it's a dedicated server. If you decide to use this library with `:enable` it's because you want it to be patched. I'd say it's not to the library to change behaviour based on server type.

> Commit message should probably be updated to describe the changes, since it appears that this PR does more than Added unit tests. It appears to drop poise and also remove some cloud and OS validation checks.

I thought this library was not used yet, so I kept the commit message minimal. I'll update it.
The poise changes and the OS checks (which are not required as the provider is OS specific already) don't change the behaviour, but you are right it worth mentioning it.

File: Rakefile (hunk 1 above)
Comment (jujugrrr) [inline Rakefile]: https://github.com/rackspace-cookbooks/phpstack/blob/master/Rakefile#L16

**Comment by jujugrrr** [comment]
@martinb3 Can you have another look ?

File: test/unit/spec/automatic_updates_ubuntu1204_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  UBUNTU1204_OPTS = {
+    log_level: LOG_LEVEL,
+    platform: 'ubuntu',
+    version: '12.04',
+    step_into: ['automatic_updates']
+  }
+
+  context 'Enable automationc_update' do
```
Comment (martinb3) [inline test/unit/spec/automatic_updates_ubuntu1204_spec.rb]: Looks like a typo on this line.

**Comment by martinb3** [comment]
I sent you a note about the commit message wrapping from the summary into the expanded commit message, and noted a typo, but LGTM. :+1: Feel free to merge.

**Pull request merged**